### ISSUE TITLE
CrabNet hyperparameter optimization benchmark competition(s)

### ISSUE THREAD

**Issue by sgbaird** [AccelerationConsortium/ac-training-lab#431]
Perhaps in its original form to start out with, with a second competition focusing on applying NChooseK-style constraints with a compositional constraint.

https://huggingface.co/spaces/AccelerationConsortium/crabnet-hyperparameter

This is one where we might just "shoot for the stars" and skip all the possible intermediates that aren't so real-world. I.e., try to match a real-world materials optimization problem as closely as possible. E.g., jump straight to high-D, multi-objective, composition constraint, cost-aware, etc.

**Comment by Amanichab** [comment]
I just wanted to make sure I'm understanding things correctly.
From what I understood, the idea would be to start with a first competition that closely follows the paper setup, and then potentially have a second, more advanced version withNChooseK-style constraints with a compositional constraint?

I'm not sure if this was meant as a confirmed plan or more of a suggestion — just wanted to check!

**Comment by Amanichab** [comment]
I guess it depends on the problem, but we could try skipping the simpler version and go straight to something more realistic — if the feedback from the basic one isn’t really needed. That said, I’m not sure if you already have a specific problem in mind and the necessary data. But if everything we need is already available, then why not!

**Comment by sgbaird** [comment]
Let's get started with the simpler one since it will make it easier to convert to the more complex one, though we may end up only advertising the more complex one.

**Comment by Amanichab** [comment]
[steps to install CrabNet on Kaggle](https://www.kaggle.com/code/amanichabouni/crabnet-installation)
[downgrade python version](https://www.kaggle.com/code/amanichabouni/python-version)

Note: This code works, but the kernel crashes every time due to the Conda environment setup.
So you'll need to run the notebook once, then manually re-run the cells after the Conda environment has been created.
TL;DR: "Run all" won't work here. You'll have to execute the cells step by step after the environment setup.
I'm still working on the kernel issue.

Also, this setup can serve as a general template for installing future Python packages that require a different Python version than the one available on Kaggle . It basically works by downgrading Python, redirecting all key pointers, and overriding the sys.path to match the new environment.

**Comment by Amanichab** [comment]
Okay ! This seems like a more stable approach.

**Comment by sgbaird** [comment]
Let's plan on warmstarting the campaign with 15 datapoints that are generated via Sobol sampling (different seed for each campaign) per comments in https://github.com/AccelerationConsortium/ac-training-lab/issues/262#issuecomment-3191636047. I think using a kaggle dataset with pregenerated points would be fine. Let's just also make sure that it's tied to a specific seed and ensure that we have a copy of the code somewhere that was used to generate that dataset.

**Comment by sgbaird** [comment]
Ah, ok. Thanks for clarifying! We can deprioritize the Ax multi-fidelity issue then
